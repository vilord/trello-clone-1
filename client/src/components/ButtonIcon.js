import React from 'react';
import PropTypes from 'prop-types';
import { Button, Icon } from 'semantic-ui-react';
import './ButtonIcon.css';

const ButtonIcon = ({ icon, label, flipped, onClick }) => (
  <Button
    className="ButtonIcon"
    icon
    onClick={onClick}
    size="large"
    style={label ? { minWidth: '90px' } : null}
  >
    <Icon name={icon} flipped={flipped ? 'horizontally' : null} /> {label}
  </Button>
);

ButtonIcon.propTypes = {
  icon: PropTypes.string.isRequired,
  label: PropTypes.string,
};

export default ButtonIcon;
