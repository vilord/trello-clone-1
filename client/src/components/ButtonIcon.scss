.ButtonIcon {
  margin: auto 4px !important;
  padding: 7px !important;

  color: white !important;
  border-radius: 3px !important;
  background-color: #5495c0 !important;

  &:hover {
    background-color: #3a85b6 !important;
  }

  &:active {
    background-color: #2176ad !important;
  }
}
