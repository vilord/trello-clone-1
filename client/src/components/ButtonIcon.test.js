import React from 'react';
import { shallow } from 'enzyme';

import ButtonIcon from './ButtonIcon';

describe('ButtonIcon component', () => {
  beforeAll(() => {
    console.error = err => {
      throw new Error(err);
    };
  });

  let component;
  beforeEach(() => {
    component = shallow(<ButtonIcon icon="" />);
  });

  it('renders', () => {
    expect(component).toHaveLength(1);
  });

  it('sets the icon based on props.icon', () => {
    const newProps = {
      icon: 'trello',
    };
    component.setProps(newProps);
    expect(component.find('Icon').prop('name')).toBe(newProps.icon);
  });

  it('sets the label based on props.label', () => {
    const newProps = {
      icon: 'trello',
      label: 'Board',
    };
    component.setProps(newProps);
    expect(
      component
        .find('Button')
        .children()
        .last()
        .text(),
    ).toBe(newProps.label);
  });
  
  it('flips the Icon when props.flipped is true', () => {
    component.setProps({ flipped: true });
    expect(component.find('Icon').prop('flipped')).toBe('horizontally');
  });

  it('calls a callback when clicked', () => {
    const onClickSpy = jest.fn();
    component.setProps({ onClick: onClickSpy });
    component.simulate('click');
    expect(onClickSpy).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    component.setProps({
      icon: 'trello',
      label: 'Board',
    });
    expect(component).toMatchSnapshot();
  });
});
