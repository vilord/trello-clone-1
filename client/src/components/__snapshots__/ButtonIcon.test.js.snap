// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon component matches snapshot 1`] = `
<Button
  as="button"
  className="ButtonIcon"
  icon={true}
  size="large"
  style={
    Object {
      "minWidth": "90px",
    }
  }
>
  <Icon
    as="i"
    flipped={null}
    name="trello"
  />
   
  Board
</Button>
`;
